### DISCUSSION TITLE
Proper restart of ModbusClientTCPasync

### DISCUSSION THREAD

**Discussion by mrmareksvk** [eModbus/eModbus#296] [Q&A] [answered]
Hello,

I am having issue with Async TCP Client in case I lose Wi-Fi connection and then reconnect.
After regaining Wi-Fi connection, server won't answer anymore to my request (nor I know if they are being sent - probably not because I can open connection to the server from another device without problem - if there is log that shows internal sending from queue I can check). 
Eventually I will get to the point when I get Error creating request: E8 - Request queue full.
From documentation I cannot see any way how to clear queue, if I try to do it via destructor and create new instance I will get ESP32 crash.

I tried multiple different approaches including:
-onErrorHandler after receiving error 234 (E9) ILLEGAL_IP_OR_PORT, will disconnect client. - MBclient.disconnect(true). And then again connect it - MBclient.connect(ip,port) when wifi is re-established.
-tried again after disconnecting - and when wifi is back on setting up onDataHandler, onErrorHandler, connect -> same result, queue will eventually show its full but non of the messages sent
-tried destructor after wifi loss, and then new instance - but this resulted into the crash.

If anybody could help me with this, or point me to right direction how to solve possible restart of TCPasync connection after Wi-Fi loss I would be glad thank you.

**Comment by bertmelis** [comment]
I'll look into the issue. I can not give a deadline though.

**Comment by schuppeste** [comment]
Same here, any Workaround?

My first Idea, restart the ESP on Disconnect. Its not the Best Solution.

**Reply by Miq1** [reply, marked as answer]
Created [an issue](https://github.com/eModbus/eModbus/issues/364) from this as @bertmelis needs to look at it.

**Reply by schuppeste** [reply]
in my Case its okay to use "ModbusClientTCP".
